hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -107,10 +106,12 @@ void rollback() {
 		if (!isInTransaction) {
 			throw new IllegalStateException("Cannot perform rollback, no transaction is active!");
 		}
-		entities.clearRefCache();
 
 		changes.clear();
 		deadEntities.clear();
+		changedKeys.clear();
+		createdKeys.clear();
+		perishedKeys.clear();
```

Comment:
Should extract these lines into a separate function, because:
1. They alone form a single functionality.
2. The extracted function will also act as a place holder for easy changing in the future.
3. Will help avoid code duplication and inconsistency when we have a need of copy-paste.